server/services/User.js

```diff
@@ -0,0 +1,39 @@
+import { randomBytes, createHmac } from 'crypto';
+import { config } from 'dotenv';
+import UserModel from '../models/User';
+
+config();
+
+const { DB_SALT } = process.env;
+
+const genRandomString = length =>
+  randomBytes(Math.ceil(length / 2))
+    .toString('hex')
+    .slice(0, length);
+
+const sha512 = (password, salt) => {
```

Comment:
lets create dedicated util file for hashing strings `/server/utils/hash.js` and move `genRandomString` + `sha512` there

---
it's better to rename `sha512` to something like `getHmac`, because sha512 is just a name of algorithm used while creating hmac